### ISSUE TITLE
Confidence score

### ISSUE THREAD

**Issue by Karlheinzniebuhr** [Wensi-Tang/OS-CNN#13]
Hi Wensi, is there a way to obtain the confidence score of predictions? I have a notebook in Keras where the model shows them. Would be useful for me to have them in os-cnn

**Comment by Wensi-Tang** [comment]
Hello, sorry for the late reply. 

I am not sure what the confidence score exactly is. I try to search "[confidence score keras](https://www.google.com/search?q=confidence+score+keras+&sxsrf=ALiCzsbqnFpEbqa4S7NoxZvsRcF6o-zlrA%3A1657241704328&ei=aIDHYvjZE_-u4-EP1saF8A0&ved=0ahUKEwj4rd-4iuj4AhV_1zgGHVZjAd4Q4dUDCA4&uact=5&oq=confidence+score+keras+&gs_lcp=Cgdnd3Mtd2l6EAMyBAgjECcyBAgjECcyBAgAEEMyCggAEIAEEIcCEBQyBQgAEIAEMgUIABCABDIFCAAQgAQyBQgAEIAEMgUIABCABDIFCAAQgAQ6BAgAEEc6BggAEB4QB0oECEEYAEoECEYYAFC6AljhCWD_CmgAcAN4AIABvAGIAe0DkgEDMC4zmAEAoAEBoAECyAEIwAEB&sclient=gws-wiz)". The top answer in this [link](https://stackoverflow.com/questions/59851961/how-to-calculate-confidence-score-of-a-neural-network-prediction) says "the softmax is a problematic way to estimate a confidence of the model`s prediction."

If you want to get the model prediction, you could check from here search "acc_test = eval_model(torch_FCN, test_loader)"
in [this](https://github.com/Wensi-Tang/OS-CNN/blob/master/1_2_OS-CNN_load_saved_model_for_prediction.ipynb) notebook. 

This [line 14](https://github.com/Wensi-Tang/OS-CNN/blob/5d606f3a32ed6c21cf67a05f4b51163625721a6a/Classifiers/OS_CNN/log_manager.py#L14) is how eval_model() works, and the [line 18](https://github.com/Wensi-Tang/OS-CNN/blob/5d606f3a32ed6c21cf67a05f4b51163625721a6a/Classifiers/OS_CNN/log_manager.py#L18) is where we get the model prediction. We could use softmax to get the confidence score.

**Comment by Karlheinzniebuhr** [comment]
Thank you, I tried softmax and it works but it is not quite what we needed I think. But we emulate probabilistic confidence by using an ensemble of several models and it works pretty good
